### ISSUE TITLE
Failed to create a pr review comment

### ISSUE THREAD

**Issue by williamfzc** [PyGithub/PyGithub#2501]
hi
I am trying to create pr review comments but getting this error:

```python
Traceback (most recent call last):
  File "/action_internal/main.py", line 173, in <module>
    main()
  File "/action_internal/main.py", line 169, in main
    send_code_comments(repo_token, after_sha, repo_name, int(issue_number), lines)
  File "/action_internal/comment.py", line 35, in send_code_comments
    pr.create_review_comment(
  File "/usr/lib/python3.10/site-packages/github/PullRequest.py", line 431, in create_review_comment
    headers, data = self._requester.requestJsonAndCheck(
  File "/usr/lib/python3.10/site-packages/github/Requester.py", line 398, in requestJsonAndCheck
    return self.__check(
  File "/usr/lib/python3.10/site-packages/github/Requester.py", line 423, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 422 {"message": "Validation Failed", "errors": [{"resource": "PullRequestReviewComment", "code": "custom", "field": "pull_request_review_thread.end_commit_oid", "message": "pull_request_review_thread.end_commit_oid is not part of the pull request"}, {"resource": "PullRequestReviewComment", "code": "missing_field", "field": "pull_request_review_thread.diff_hunk"}], "documentation_url": "https://docs.github.com/rest/reference/pulls#create-a-review-comment-for-a-pull-request"}
```

It seems that some object defs were out-of-dated.

https://github.com/PyGithub/PyGithub/blob/300c5015d41236eb410b9cbe895258294e3500af/github/PullRequest.py#L425

```python
post_parameters = {
            "body": body,
            "commit_id": commit_id._identity,
            "path": path,
            "position": position,
        }
```

https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#create-a-review-comment-for-a-pull-request

The official doc said `position` is deprecated. Use line instead.

**Comment by heitorpolidoro** [comment]
Hi,

I created this pull request #2540 that might fix this issue

**Comment by EnricoMi** [comment]
This has been fixed in #2540.
